designs/spot-consolidation.md

```diff
@@ -0,0 +1,106 @@
+# Spot Consolidation
+
+## Problem Statement
+
+Customers face increased costs due to underutilized spot instances. A customer with 4 x 24cpu pods would launch a 24xl (96cpu) instance type. If three of the four pods complete, the instance will indefinitely run at 25% utilization until the pod completes, is able to fit elsewhere in the cluster, or additional work is scheduled to the node. Customers have raised this inefficiency as a critical pain point with Karpenter: https://github.com/aws/karpenter-core/issues/763. 
```

Comment:
Why are there only three options:

1. Wait for remaining pod to complete in order to GC spot node.
2. Rely on another compute slot on another, existing spot node to accommodate a gracefuol pod migration, then delete the original, larger spot node.
3. Wait for more spot workloads to arrive to remove overhead inefficiencies on larger, currently underutilized spot node.

And not this one as well:

4. Build a new, smaller spot node to run the remaining workload, migrate workload, and delete the original, larger spot node.

?

Reply:
We refer to 4 as "replacement", and is the only thing in scope of this document, discussed in the second paragraph.

> If Karpenter were to consolidate spot capacity based purely on cost, it would repeatedly interrupt the same instance, walking down the PCO decision ladder, until only the lowest price remained. Because of these complications, Karpenter does not currently support "spot replacement" consolidation.

I can clarify this in the language.